src/main/java/com/gtnewhorizons/angelica/transform/HUDCachingTransformer.java

```diff
@@ -0,0 +1,75 @@
+package com.gtnewhorizons.angelica.transform;
+
+import com.gtnewhorizons.angelica.loading.AngelicaTweaker;
+import net.minecraft.launchwrapper.IClassTransformer;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AnnotationNode;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.InsnList;
+import org.objectweb.asm.tree.InsnNode;
+import org.objectweb.asm.tree.JumpInsnNode;
+import org.objectweb.asm.tree.LabelNode;
+import org.objectweb.asm.tree.LdcInsnNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+import org.objectweb.asm.tree.MethodNode;
+
+
+public class HUDCachingTransformer implements IClassTransformer {
+    private static final String HUDCaching = "com/gtnewhorizons/angelica/hudcaching/HUDCaching$Hooks";
+
+    @Override
+    public byte[] transform(String name, String transformedName, byte[] basicClass) {
+
+        final ClassReader cr = new ClassReader(basicClass);
```

Comment:
Why are you parsing all the classes in the game if it's only going to have effect on one thaumcraft class ?

Reply:
Because more will be added in the future.

Reply:
Changed the ASM logic to only inject into methods that we have listed. List is now in the transformer and more will be added in the future.
https://github.com/GTNewHorizons/Angelica/pull/562/files#diff-ad067efcbda759d7dfa415c8453cf8708493c9e2d9a06f00da83728caa449948R23-R25

Reply:
you are still reading and parsing all the classes in the game only to change one class, please just check at the start of the method : 
`
if (!"thaumcraft.client.lib.RenderEventHandler".equals(transformedName)) {
 return bytes;
}
`

Later on if you want to add more classes and methods, you can create a Map<String, List\<String\>> and use as a key the class names and as a value the method names to transform